prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -266,28 +265,33 @@ protected APIResult delete(String type, String entity, String colo) {
         try {
             EntityType entityType = EntityType.getEnum(type);
             String removedFromEngine = "";
+            Entity entityObj;
             try {
-                Entity entityObj = EntityUtil.getEntity(type, entity);
-                verifySafemodeOperation(entityObj, EntityUtil.ENTITY_OPERATION.DELETE);
+                entityObj = EntityUtil.getEntity(type, entity);
+            } catch (FalconException e) {
```

Comment:
This is no different from what is happening currently. org.apache.falcon.entity.EntityUtil#getEntity(org.apache.falcon.entity.v0.EntityType, java.lang.String) throws a EntityNotRegisteredException which is a sub-class of FalconException.